source/computer/commands.rst

```diff
@@ -72,45 +72,565 @@ Linux/macOS 下有成百上千个命令，每个命令都有众多选项。这
+
+    $ sort -t ":" -k 2 seismo-learn-sort.txt
+
+同上，但使用降序排序（以下两种方式均可）::
+
+    $ sort -t ":" -k 2r seismo-learn-sort.txt
+
+同上，但从第二个字段的第三个字符到行末，降序排序::
+
+    $ sort -t ":" -k 2.3r seismo-learn-sort.txt
+
+同上，但从第二个字段的第三个字符开始到第三个字段的第四个字符结束，降序排序::
+
+    $ sort -t ":" -k 2.3,3.4r seismo-learn-sort.txt
+
+只按照第二个字段进行降序排序::
+
+    $ sort -t ":" -k 2,2r seismo-learn-sort.txt
+
+还可以多次使用 ``-k`` 选项，这样会在前一个选项指定的字符范围的排序结果相同时，
+接着比较下一个选项指定的字符范围。
+
+按照第二个字段进行降序排序，若相同则按照第三个字段升序排序::
+
+    $ sort -t ":" -k 2,2r -k 3,3 seismo-learn-sort.txt
+
+按照第二个字段进行降序排序，若相同则按照第四个字段的数值大小降序排序::
+
+    $ sort -t ":" -k 2,2r -k 4,4nr seismo-learn-sort.txt
+
+同上，但并忽略相同行::
+
+    $ sort -t ":" -k 2,2r -k 4,4nr -u seismo-learn-sort.txt
+
 ssh
 ---
 
+``ssh`` 命令的命名源于 **S**\ ecure **Sh**\ ell（安全外壳协议，简称 SSH），该协议是
+一种加密的网络传输协议。使用 ``ssh`` 命令可以登录到远程电脑中。常用于登录服务器提交计算任务。
+
+若远程电脑的 IP 地址是 192.168.1.100，用户名是 seismo-learn，执行以下命令可以登录到该
+电脑中::
+
+    $ ssh seismo-learn@192.168.1.100
+
 tail
 ----
 
+``tail`` 命令用于打印文件尾部内容。默认打印文件的后 10 行。
+
+打印 :file:`/etc/passwd` 的后 10 行::
+
+    $ tail /etc/passwd
+
+打印 :file:`/etc/passwd` 的后 5 行::
+
+    $ tail -n 5 /etc/passwd
+
+如果 ``-n`` 选项后的数字（N）前面有加号 :kbd:`+`，则打印的是文件开头的第 N 行到文件末尾，
+并不是最后 N 行。例如，打印 :file:`/etc/passwd` 的第 5 行到文件末尾::
+
+    $ tail -n +5 /etc/passwd
+
 tar
 ---
 
+``tar`` 命令的名字来自 **t**\ ape **ar**\ chive（磁带存档），因为该命令最初被用来在磁带上
+创建档案。该命令可以把一大堆文件和目录打包成一个文件，并且可以选择压缩该文件，这对于备份文件或
+将几个文件组合成一个文件以便于网络传输是非常有用的。
+
+首先要弄清两个概念：打包和压缩。打包是指将一大堆文件或目录打包成一个文件，而压缩则是将一个大文件
+通过一些压缩算法变成一个小文件。Linux 中的很多压缩程序只能对压缩单个文件，若想压缩一大堆文件，
+首先得将这一大堆文件打成一个包（使用 ``tar`` 命令），再用压缩程序进行压缩
+（使用 ``gzip`` 或 ``bzip2`` 命令）。使用 ``tar`` 命令时，可以直接选择压缩打包的文件，无需
+再单独使用压缩程序进行压缩。
+
+将 :file:`file1` 和 :file:`file2` 文件打包，打包文件命名为 :file:`seismo-learn.tar`
+（选项 ``-f`` 后的打包文件名一般用 :file:`.tar` 来作为标识）::
+
+    $ tar -cvf seismo-learn.tar file1 file2
+
+同上，但在打包后用 ``gzip`` 命令进行压缩（一般用 :file:`.tar.gz` 或 :file:`.tgz` 来作标识）::
+
+    $ tar -zcvf seismo-learn.tar.gz file1 file2
+
+打包后，用 ``bzip2`` 命令进行压缩（一般用 :file:`.tar.bz2` 或 :file:`.tbz` 来作标识）::
+
+    $ tar -jcvf seism-learn.tar file1 file2
+
+打包和压缩 :file:`file1` 和 :file:`file2` 文件:
+
+.. tabs::
+
+   .. code-tab:: bash 仅打包
+
+      # 打包文件命名为 seismo-learn.tar（选项 -f 后的打包文件名一般用 .tar 来作为标识）
+      $ tar -cvf seismo-learn.tar file1 file2
+
+   .. code-tab:: bash 打包后用 gzip 命令压缩
+
+      # 打包压缩文件命名为 seismo-learn.tar.gz（选项 -f 后的打包压缩文件名一般用 .tar.gz 或 .tgz 来作为标识）
+      $ tar -zxvf seismo-learn.tar.gz
+
+   .. code-tab:: bash 打包后用 bzip2 命令压缩
+
+      # 打包压缩文件命名为 seismo-learn.tar.bz2（选项 -f 后的打包压缩文件名一般用 .tar.bz2 或 .tbz 来作为标识）
+      $ tar -jxvf seismo-learn.tar.bz2
+
+查阅打包压缩文件含有哪些文件和目录::
+
+    $ tar -tvf seismo-learn.tar
+    $ tar -tvf seismo-learn.tar.gz
+    $ tar -tvf seismo-learn.tar.tb2
+
+将打包文件 :file:`seismo-learn.tar` 还原::
+
+    $ tar -xvf seismo-learn.tar
+
+将打包压缩文件 :file:`seismo-learn.tar.gz` 解压还原::
+
+    $ tar -zxvf seismo-learn.tar.gz
+
+将打包压缩文件 :file:`seismo-learn.tar.bz2` 解压还原::
+
+    $ tar -jxvf seismo-learn.tar.bz2
```

Comment:
I think `tar` is very clever in detecting the compression format, so `tar -xvf` should work for both `tar.gz` and `tar.bz2`. Please confirm it.

Reply:
will try it.

Reply:
Yes, `tar -xvf` works for all.